Makefile

```diff
@@ -0,0 +1,66 @@
+.PHONY: update_riot_stats update_riot_repo update_riot_board_list update_riot_data
+.PHONY: update_riot_contributors install_python_requirements
+.PHONY: update_riot_drivers update_riot_cpus
+.PHONY: build serve
+
+RIOT_REPO_URL = "https://github.com/RIOT-OS/RIOT.git"
+
+TOOLS_DIR = $(abspath _tools)
+DATA_DIR = $(abspath _data)
+_DEFAULT_RIOTBASE = $(abspath _RIOT)
+RIOTBASE ?= $(_DEFAULT_RIOTBASE)
+
+RIOT_BOARDS_FILE = $(DATA_DIR)/riot_boards.yml
+RIOT_STATS_FILE = $(DATA_DIR)/riot_stats.yml
+RIOT_CONTRIBUTORS_FILE = $(DATA_DIR)/contributors.json
+RIOT_DRIVERS_FILE = $(DATA_DIR)/riot_drivers.csv
+RIOT_DRIVERS_CATS_FILE = $(DATA_DIR)/riot_drivers_cats.csv
+RIOT_CPUS_FILE = $(DATA_DIR)/riot_cpus.yml
```

Comment:
we were thinking with Leandro to leave some as a stub (so one doesn't need to call the update target). But this would be indeed possible.

What do you think?

Reply:
I think that if someone generates the website locally to get a local preview, all these files will end up modified (in git), so this will be annoying.
Better remove them.

Reply:
But one could build the website locally without calling `make update_riot_data`, then the files would not be updated and the stub values would be used instead.

Reply:
but we can add a `.gitignore` and then those files would still be there, but won't be tracked.

Reply:
For newcomers, the documentation says that `make build` and/or `make serve` should be used. And these commands will call the `update_riot_data` anyway.

Reply:
> but we can add a .gitignore and then those files would still be there, but won't be tracked.

Sorry, my [initial comment](https://github.com/RIOT-OS/riot-os.org/pull/17#discussion_r582946488) was maybe not explicit enough.

Reply:
> For newcomers, the documentation says that `make build` and/or `make serve` should be used. And these commands will call the `update_riot_data` anyway.

No, they don't `update_riot_data`. It was intentional so the file would not end up modified as you pointed out.

Reply:
> It was intentional so the files would not end up modified as you pointed out.

Sorry, I didn't understand it like that. Regenerating the files for each build (to get the latest data) and putting these files in .gitignore seems cleaner. But maybe there's a drawback doing so ?

Reply:
I will try it out and see how it looks like

Reply:
> Regenerating the files for each build (to get the latest data) and putting these files in .gitignore seems cleaner. But maybe there's a drawback doing so ?

I can think of someone trying to build the website while offline? So why force them to update the data on each build?

Reply:
long story short: the checkout of the website should provide a stub of the generated meta data, because we don't require that the user who edits the webpage is required to build the data. if the user manually updates the generated data locally, these changes should not be pushed.

Reply:
> So why force them to update the data on each build?

But the generated data are taken from the local RIOT clone ? so that could be done offline.

Reply:
> > So why force them to update the data on each build?
> 
> But the generated data are taken from the local RIOT clone ? so that could be done offline.

If there user has `RIOTBASE` set to the local repo or has already run the update command then it would have it locally, yes. But the contributor list is always fetched from the GH API.

Reply:
my proposal didn't work because -gitignore only works for untracked files :(. However, I solved in a more standard way in the last pushups (regular `dist` files that get copied if there's no data)

Reply:
> However, I solved in a more standard way in the last pushups (regular dist files that get copied if there's no data

Doesn't convince me. The URL of boads/cpu/drivers will change if one day we change the documentation generator (to sphinx maybe).
To solve the offline local build problem, since it seems that only the request to contributors is remaining, the Python script that fetches this information could be adapted as follows:
- use a try/except around the request to the github api
- in case of a connection failure, look for any existing contributors.json file, if it exists, do nothing
- if contributors.json doesn't exist, print a warning and write dummy data in it (a single fake contributor)

Reply:
> Doesn't convince me. The URL of boads/cpu/drivers will change if one day we change the documentation generator (to sphinx maybe), so the content of these files will be outdated at that point.

The thing is, the stub is just a snapshot of `make update_riot_data`. If at some point the documentation engine changes, we will need to change `make update_riot_data` anyway (and therefore re-generate these stubs). So the problem is there even if we don't use stubs :/

> To solve the offline local build problem, since it seems that only the REST request to fetch contributors is remaining, the Python script that fetches this information could be adapted as follows:

The only reason IMO to keep the stubs there is because `make update_riot_data` take some time:

```bash
~/Development/riot-os.org/ hyperfine "make update_riot_data"
Benchmark #1: make update_riot_data
  Time (mean ± σ):     36.466 s ±  2.093 s    [User: 99.641 s, System: 3.350 s]
  Range (min … max):   32.159 s … 40.039 s    10 runs
```

I personally never understood it as a way to bypass building offline.
I can see that we could get this information directly from the Github API, but since we are already generating stubs for the other files, what would be the benefit of doing that at this point?

Reply:
> > Doesn't convince me. The URL of boads/cpu/drivers will change if one day we change the documentation generator (to sphinx maybe), so the content of these files will be outdated at that point.
>
> The thing is, the stub is just a snapshot of make update_riot_data. If at some point the documentation engine changes, we will need to change make update_riot_data anyway (and therefore re-generate these stubs). So the problem is there even if we don't use stubs :/

we don't need to make things more complicated than necessary. that we need to update the scripts when we change the infrastructure, for example, to create docs, is not a problem at all.

Reply:
> So the problem is there even if we don't use stubs :/

It's different: you just have to update the script(s), not both the script and the "stub" data.

> The only reason IMO to keep the stubs there is because make update_riot_data take some time

This doesn't necessarily have to be run for every build. At least once, the first time you try to build, to bootstrap the local data. After that it can be updated from time to time but there's no absolutely no need to commit these files in the repo.
In your Makefile, you eventually make these files a dependency of the corresponding make targets and keep a separate update_riot_data target that forces the update, or something like that.
My main issue is to keep these outdated "stub" data in the repo. It's pretty clear that they are not needed.

Reply:
> we don't need to make things more complicated than necessary

This is exactly my point to which I add unnecessary cluttering of the repository.

Reply:
> My main issue is to keep these outdated and generated "stub" data in the repo. It's pretty clear that they are not needed.

i don't understand this. when they are empty and i don't run `make update_riot_data` i don't see anything.

Reply:
Sorry if it was unclear, but I can rephrase with more details: my main issue is to keep outdated and generated "stub" data committed in the repo. It's pretty clear that generated data shouldn't be committed.

So to summarize, this is the following behavior I have in mind (when running make build or make preview):
- if the files containing generated data are not there already (first time build/preview on a freshly cloned repository), generate them. This can be done with the make file dependency mechanism.
- if they are already there, do not fetch/update, just build
- if they need an update, call `make update_riot_data` manually before.

In the CI, a simple `make build` would fetch the data and create the file in one call.
For a regular user, they are fetched the first build time (like for the CI), but after, we provide helpers to update from time to time.
In any case, having these files committed in the repository is required so these committed files as-is now are just cluttering the repository.

Reply:
[`make build`]
> if the files containing generated data are not there already (first time build/preview on a freshly cloned repository), generate them. This can be done with the make file dependency mechanism.

i disagree. there might be website editors that run `make build` but don't care about fresh data. we don't want to force website editors to update such data because it introduces additional (unnecessary) steps.

Reply:
> it introduces additional (unnecessary) steps

As I said, it doesn't introduce additional steps for the user, since it's automatically handled by `make build` the first time.

> we want to provide a 'minimal' running version that might include outdated data but shows the content elements.

Then why all these stub files are taken from the generator directly. If only the _content elements_ are important (whatever this means), you could simply have dummy data (like fake data), that wouldn't necessarily come from the RIOT website. I mean there's no need to commit a 400 lines files containing all boards supported in February 2021 or a **2k** file containing all contributors (and also containing unnecessary information) . Let's also mention this one line json file of **20k** columns for the drivers! And this is without mentioning the mixed formats (some in json, some in yml and, with this PR, some in csv). This is just cluttering the repository (repeating myself). 

Again, since they are generated, they shouldn't be committed in the repo.

Reply:
It sounds like having stub files with a subsets of the lists is a good tradeoff between having empty pages when not updating, and cluttering the repo with huge files.  But in this case `make build` would not have to trigger an update, right? Should we proceed with that?

Reply:
> Then why all these stub files are taken from the generator directly. If only the content elements are important (whatever this means), you could simply have dummy data (like fake data), that wouldn't necessarily come from the RIOT website. 

but it doesn't harm either. taking a snapshot and use this as dummy data is completely fine. ... anyhow, as @leandrolanzieri suggested, he will include a reduced set of data to spare some space.

i think we can proceed.

Reply:
> It's pretty clear that generated data shouldn't be committed.

Note that the only reason why we were doing this was for generating dummy files. Sorry if it was not clear

Reply:
as discussed, I reduced the stub files. I chose some numbers that allow to test search functionalities and responsiveness.